src/cosmos-copilot.WebApp/Services/ChatService.cs

```diff
@@ -246,23 +220,19 @@ private int GetTokens(string userPrompt)
         Tokenizer _tokenizer = Tokenizer.CreateTiktokenForModel("gpt-3.5-turbo");
 
         return _tokenizer.CountTokens(userPrompt);
+
     }
+   
 
     /// <summary>
     /// Query the semantic cache with user prompt vectors for the current context window in this conversation
     /// </summary>
-    private async Task<(string cachePrompts, float[] cacheVectors, string cacheResponse)> GetCacheAsync(List<Message> contextWindow)
+    private async Task<string> GetCacheAsync(float[] vectors)
     {
-        //Grab the user prompts for the context window
-        string prompts = string.Join(Environment.NewLine, contextWindow.Select(m => m.Prompt));
-
-        //Get the embeddings for the user prompts
-        float[] vectors = await _semanticKernelService.GetEmbeddingsAsync(prompts);
 
         //Check the cache for similar vectors
-        string response = await _cosmosDbService.GetCacheAsync(vectors, _cacheSimilarityScore);
+        return await _cosmosDbService.GetCacheAsync(vectors, _cacheSimilarityScore);
```

Comment:
If this function is now one line calling the `GetCacheAsync()` from the Cosmos service, should we just call it directly instead of also having the `GetCacheAsync()` helper method in the Chat service?

Reply:
I thought about that too but it seemed more symmetrical to have both calls to the cache calling a local function first. I'm fine either way you want to do this since I know this is going to be in the lab guide.